### DISCUSSION TITLE
Anyone with a good How to install on Ubuntu 22

### DISCUSSION THREAD

**Discussion by stccorp** [imartinez/privateGPT#1275] [General]
Hello,

I tried following the instructions and nothing work.  I am running Ubuntu 22 and Conda . Thank you

For example 
poetry install --with ui,local

  Stack trace:

  11  /usr/lib/python3/dist-packages/clikit/console_application.py:123 in run
      io = io_factory(
a bunch of errors
  /usr/lib/python3/dist-packages/clikit/args/default_args_parser.py:247 in _parse_long_option
      self._add_long_option(name, None, tokens, fmt, lenient)

  NoSuchOptionException

  The "--with" option does not exist.

**Comment by FourEyes4456** [comment]
I've done this about 10 times over the last week, got a guide written up for exactly this

#Install Linux

sudo apt update
sudo apt-get install build-essential procps curl file git -y
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"' >> $HOME/.bash_profile
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
brew doctor
#Should respond with "Your system is ready to brew."
brew install GCC
sudo apt install python3.11 -y
sudo apt install python3-pip
sudo apt install pipx -y
pipx install poetry
pipx ensurepath
brew install make

#Close and Re-Open Terminal

git clone https://github.com/imartinez/privateGPT
cd privateGPT
poetry install --with ui
poetry install --with local
poetry run python scripts/setup

make run

**Reply by kladskull** [reply]
This is quite interesting.. does brew work on Ubuntu? (Not that I ever would..)

**Reply by FourEyes4456** [reply]
It seems like it does, yeah. Ran that a few months back and I've had next to no problems with it

**Comment by mainframecn** [comment]
The "poetry install --with ui" and "poetry install --with local" don't work for me anymore. I think it's now "poetry install --extras "ui" local seems to have gone missing.
